protos/file2.proto

```diff
@@ -0,0 +1,206 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+syntax = "proto3";
+
+package lance.file.v2;
+
+import "google/protobuf/any.proto";
+
+// # Lance v0.2 File Format
+//
+// The Lance file format is a barebones format for serializing columnar data
+// into a file.
+//
+// * Each Lance file contains between 0 and 4Gi columns
+// * Each column contains between 0 and 4Gi pages
+// * Each page contains between 0 and 4Gi items
```

Comment:
> 4Gi limitation because we use uint32 for anything?

Yes.  Also, there is just no good reason to have more than 4Gi items in a page.

> We don't track byte offset in this case?

Byte offsets appear in a few places (e.g. footer) but they are all u64.